vector/src/test/java/im/vector/app/features/spaces/SpaceListViewModelTest.kt

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.spaces
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.airbnb.mvrx.Success
+import com.airbnb.mvrx.test.MvRxTestRule
+import im.vector.app.features.analytics.plan.Interaction
+import im.vector.app.test.fakes.FakeAnalyticsTracker
+import im.vector.app.test.fakes.FakeAutoAcceptInvites
+import im.vector.app.test.fakes.FakeRoomService
+import im.vector.app.test.fakes.FakeSession
+import im.vector.app.test.fakes.FakeSessionAccountDataService
+import im.vector.app.test.fakes.FakeSpaceService
+import im.vector.app.test.fakes.FakeSpaceStateHandler
+import im.vector.app.test.fakes.FakeUserService
+import im.vector.app.test.fakes.FakeVectorPreferences
+import im.vector.app.test.fixtures.RoomSummaryFixture.aRoomSummary
+import im.vector.app.test.fixtures.UserFixture.aUser
+import im.vector.app.test.test
+import io.mockk.every
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.matrix.android.sdk.api.session.room.model.RoomSummary
+import org.matrix.android.sdk.api.util.MatrixItem
+
+internal class SpaceListViewModelTest {
```

Comment:
I couldn't add unit tests for anything that involved the RoomSummaries call in this class as it gets it as a `PagedList`. I can't figure out why but mocking a paged list simply doesn't work (when having a mock paged list in a LiveData, the LiveData still appears empty)